### ISSUE TITLE
Wrong OSCR curve for OpenMax algorithm?

### ISSUE THREAD

**Issue by linzhiqiu** [Vastlab/Reducing-Network-Agnostophobia#1]
I am confused by the difference between the OSCR curve of SoftMax threshold and OpenMax algorithm. To me, since the OSCR curve only takes probability score on training classes into account, the softmax threshold and open max algorithms should essentially behave the same under this metric. What causes the difference? And why does the Softmax Threshold curve has a big cliff as shown in the plot in paper (yet OpenMax has a smoother curve)?

**Comment by akshay-raj-dhamija** [comment]
Hello, Thanks for your interest in this work.

The scores from openmax are different from softmax as they are adjusted by the abating weibull distribution created using the deep features for each class. Since openmax is simply a post-processing step on a softmax network you may consider Fig 1(a) for a better understanding. When openmax creates weibulls on the deep features of the knowns, the overlap of knowns and unknowns in this feature space results in the unknowns being provided very high probability score of being one of the known classes. Since most of the unknowns have a higher probability score than the knowns the OSCR has the big cliff.

Another way to understand this would be that for softmax the more a samples representation moves further away from the origin of the 2D feature space the higher would be its score of being one of the known classes whereas this won't be true for openmax. For openmax the maximum score for knowness may be present at the mean of the feature representations of the training samples for a given class.

Hope this helps!

**Comment by akshay-raj-dhamija** [comment]
Regarding the big cliff in the softmax curve, it is because softmax provides a score of 1.0 to most of the unknown and known samples. This is the reason till FPR of 10^-2 the CCR is 0. According to the CCR value at FPR 10^-2, 60% of knowns also have a score of 1.0. OpenMax due to its abating probability seems to be providing a bit smoother curve.

**Comment by akshay-raj-dhamija** [comment]
Closing this for now! Please reopen for a followup.

**Comment by linzhiqiu** [comment]
Thanks for your detailed response! It's really helpful for understanding this metric. I have one more question - why the paper only compares with the AUC/Precision recall curve, but not AUC/receiver operating characteristic (ROC) curve? I believe the latter one is more commonly used, and shouldn't have the problems of the PR curve as mentioned in paper because it's calculation is similar to the OSCR metric, just that the ROC curve doesn't plot the multi-class accuracy on y-axis. But arguably OSCR is more interesting since it conveys more information in a single plot. Is my understanding right?

**Comment by akshay-raj-dhamija** [comment]
You are right! In my opinion, the evaluation for an open set problem needs to be considered both as a multiclass problem for evaluating the knowns and a binary class for evaluating how many unknowns were misclassified as knowns. Since the true positive rate on the y-axis in a ROC curve is not the same as the accuracy for a multiclass problem it is hard to convey all the information with ROC. As mentioned in the paper the OSCR is very similar to the DIR curve used in biometrics rather than ROC.
